llvm/test/Transforms/ConstraintElimination/uscmp.ll

```diff
@@ -0,0 +1,110 @@
+; NOTE: Assertions have been autogenerated by utils/update_test_checks.py
+; RUN: opt -passes=constraint-elimination -S %s | FileCheck %s
+
+define i8 @scmp_1(i32 %x, i32 %y) {
+; CHECK-LABEL: @scmp_1(
```

Comment:
Could. you also add some negative tests?

Reply:
Added negative tests for mismatched signedness of predicates. I can't think of any other things that we'd need to test that aren't already handled by other tests targeting the constraint elimination algorithm specifically rather than just its use for `ucmp`/`scmp` intrinsics.